src/lib/expansion/expansion-panel.ts

```diff
@@ -78,6 +80,18 @@ export class MatExpansionPanel extends CdkAccordionItem
   }
   private _hideToggle = false;
 
+  /** An event emitted before the body's expansion animation happens. */
+  @Output() beforeExpanded = new EventEmitter<null>();
+
+  /** An event emitted after the body's expansion animation happens. */
+  @Output() afterExpanded = new EventEmitter<null>();
+
+  /** An event emitted before the body's collapse animation happens. */
+  @Output() beforeCollapsed = new EventEmitter<null>();
+
+  /** An event emitted after the body's collapse animation happens. */
+  @Output() afterCollapsed = new EventEmitter<null>();
```

Comment:
Any thoughts on e.g. `afterExpand` vs. `afterExpanded`?

Reply:
Updated to match the other usages in the library